extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,236 @@
+
+public class GeometryUtils
+{
+
+	private static Logger LOGGER = Logger.getLogger(GeometryUtils.class);
+
+	/**
+	 * Build a buffer around a geometry for the pro
+	 * 
+	 * @param crs
+	 * @param geometry
+	 * @param distanceUnits
+	 * @param distance
+	 * @return
+	 * @throws TransformException
+	 */
+	public static final Pair<Geometry, Double> buffer(
+			final CoordinateReferenceSystem crs,
+			final Geometry geometry,
+			final String distanceUnits,
+			final double distance )
+			throws TransformException {
+		Unit<?> unit;
+		try {
+			unit = (Unit<?>) new UnitBinding().parse(
+					null,
+					distanceUnits);
+		}
+		catch (final Exception e) {
+			unit = SI.METER;
+			LOGGER.warn(
+					"Cannot lookup unit of measure " + distanceUnits,
+					e);
+		}
+		final double meterDistance = unit.getConverterTo(
+				SI.METER).convert(
+				distance);
+		final double degrees = distanceToDegrees(
+				crs,
+				geometry,
+				meterDistance);
+		// buffer does not respect the CRS; it uses simple cartesian math.
+		return Pair.of(
+				adjustGeo(
+						crs,
+						geometry.buffer(degrees)),
+				degrees);
+
+	}
+
+	/**
+	 * Fix the polygon so that long/lat fits into the bounds.
+	 * 
+	 * @param geometry
+	 * @return
+	 */
+	public static Geometry adjustGeo(
+			final CoordinateReferenceSystem crs,
+			final Geometry geometry ) {
+
+		final Coordinate[] geoCoords = geometry.getCoordinates();
+		final Coordinate[] newCoords = new Coordinate[geoCoords.length];
+
+		int i = 0;
+		for (Coordinate geoCoord : geoCoords) {
+			newCoords[i++] = clipRange(
+					crs,
+					geoCoord);
+		}
+		if (geometry instanceof Point) {
+			return geometry.getFactory().createPoint(
+					newCoords[0]);
+		}
+		else if (geometry instanceof LineString) {
+			return geometry.getFactory().createLineString(
+					newCoords);
+		}
+		return geometry.getFactory().createPolygon(
+				newCoords);
+	}
+
+	public static Coordinate clipRange(
+			final CoordinateReferenceSystem crs,
+			final Coordinate coord ) {
+		return new Coordinate(
+				clipRange(
+						coord.x,
+						crs,
+						0),
+				clipRange(
+						coord.y,
+						crs,
+						1),
+				clipRange(
+						coord.z,
+						crs,
+						2));
+	}
+
+	/**
+	 * This is perhaps a brain dead approach to do this, but it does handle wrap
+	 * around cases
+	 * 
+	 * @param val
+	 * @param bound
+	 * @return
+	 */
+	private static double clipRange(
+			final double val,
+			final CoordinateReferenceSystem crs,
+			int dimension ) {
+		final CoordinateSystem coordinateSystem = crs.getCoordinateSystem();
+		if (coordinateSystem.getDimension() > dimension) {
+			double lowerBound = coordinateSystem.getAxis(
+					dimension).getMinimumValue();
+			double bound = coordinateSystem.getAxis(
+					dimension).getMaximumValue() - lowerBound;
+			double sign = sign(val);
+			// re-scale to 0 to n, then determine how many times to 'loop
+			// around'
+			double mult = Math.floor(Math.abs((val + sign * (-1.0 * lowerBound)) / bound));
```

Comment:
is this assuming the absolute value of lowerBound is essentially upperBound?  Works for longitude so I don't care and in any case I don't think it really matters, I'd just suggest forgoing adjustGeo().